### PR TITLE
Use static linking for the binaries

### PR DESCRIPTION
**Pull request by abeltay**

### CODE REVIEW

File: hack/build-all.bash [hunk 1]
```diff
@@ -25,8 +25,8 @@ mkdir -p release
 for OS in ${BUILD_PLATFORMS[@]}; do
   for ARCH in ${BUILD_ARCHS[@]}; do
     echo "Building for $OS/$ARCH"
-    GOARCH=$ARCH GOOS=$OS CGO_ENABLED=1 $GO_BUILD_CMD -ldflags "$GO_BUILD_LDFLAGS"\
-     -o "release/sql-migrate-$OS-$ARCH" .
+    GOARCH=$ARCH GOOS=$OS CGO_ENABLED=1 $GO_BUILD_CMD --ldflags '-extldflags "-static"' \
```
Comment (abeltay) [inline hack/build-all.bash]: It's a little odd, but putting `--ldflags '-extldflags "-static"'` inside `-ldflags` will cause the build to fail.

Comment (soedar) [reply hack/build-all.bash]: oh it's interesting, the deleted line is `-ldflags` instead of `--ldflags`. hmm

Comment (soedar) [reply hack/build-all.bash]: does it work if you put `-extldflags "-static"` inside $GO_BUILD_LDFLAGS?

Comment (abeltay) [reply hack/build-all.bash]: `flag provided but not defined: -extldflags "-static"`

**Comment by abeltay** [comment]
This does not work on Linux, which travis is built on. So, I'll be closing this PR.

**Pull request closed without merging**